### PR TITLE
Reference input added to invoice generate

### PR DESCRIPTION
**Pull request by abinashpa**
[**Notion card**](https://www.notion.so/saeloun/The-user-should-be-able-to-add-references-while-generating-an-invoice-de5b8430ae304663bc045ee48d2a53a7)

## Summary
- Input field for reference added in generate invoice page

## Preview
Screenshot 2022-10-20 at 1 57 36 PM

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to
      not work as expected)
- [ ] This change requires a documentation update

### Checklist:

- [x] I have manually tested all workflows
- [x] I have performed a self-review of my own code
- [ ] I have added automated tests for my code

### CODE REVIEW

**Review by apoorv1316 — CHANGES_REQUESTED** [review]
@abinashpa Reference is not there on the invoice email and edit invoice page. Also, as per ticket reference no. can't be longer than 12 characters but that implementation is missing(see image 3)
Screenshot 2022-10-21 at 11 45 01 AM
Screenshot 2022-10-21 at 11 49 22 AM
Screenshot 2022-10-21 at 11 53 06 AM

**Comment by apoorv1316** [comment]
@abinashpa Reference is still missing on the invoice email 
Screenshot 2022-10-28 at 11 06 33 AM

**Review by prasanthchaduvula — CHANGES_REQUESTED** [review]
@abinashpa Please add some test cases and If the user has added a reference number while generating the invoice then, it should be visible to the client on the invoice email and invoice PDF.  For this you have to pdf generation file and pdf view

**Comment by apoorv1316** [comment]
> @abinashpa Reference is still missing on the invoice email Screenshot 2022-10-28 at 11 06 33 AM

@abinashpa Is this fixed?

**Comment by abinashpa** [comment]
> > @abinashpa Reference is still missing on the invoice email Screenshot 2022-10-28 at 11 06 33 AM
> 
> @abinashpa Is this fixed?

Not yet, I have an issue with elastic-search in my local.

**Comment by abinashpa** [comment]
@apoorv1316  Reference added mailer.
Screenshot 2022-11-03 at 12 22 51 AM

File: app/javascript/src/components/Invoices/common/InvoiceDetails/index.tsx [hunk 1]
```diff
@@ -86,12 +87,12 @@ const InvoiceDetails = ({
...
-          Reference
-        </p>
-        <p className="font-normal text-base text-miru-dark-purple-1000">
-          {reference}
-        </p>
+        <div className="flex flex-col">
+          <p className="font-normal text-xs text-miru-dark-purple-1000 mt-4">
+            Reference
+          </p>
+          <input type="text" value={reference} onChange={(e) => setReference(e.target.value)} className="px-2 w-3/5"/>
```
Comment (prasanthchaduvula) [inline app/javascript/src/components/Invoices/common/InvoiceDetails/index.tsx:94]: @abinashpa Please make the indentation more readable and scalable
```
<input 
  type="text" 
  value={reference} 
  onChange={(e) => setReference(e.target.value)} 
  className="px-2 w-3/5"
  />
```

**Review by prasanthchaduvula — CHANGES_REQUESTED** [review]
@abinashpa Please add test cases for reference like allowing invoice with or without reference, max length allowed for invoice sort of conditions

**Review by apoorv1316 — APPROVED** [review]

**Comment by abinashpa** [comment]
@prasanthchaduvula Tests have been incorporated.

**Review by prasanthchaduvula — APPROVED** [review]

**Pull request merged**